pkg/operator/ceph/controllers/nodedrain/reconcile.go

```diff
@@ -0,0 +1,119 @@
+	scheme  *runtime.Scheme
+	client  client.Client
+	log     *capnslog.PackageLogger
+	options *controllerconfig.Options
+}
+
+// Implement reconcile.Reconciler so the controller can reconcile objects
+var _ reconcile.Reconciler = &ReconcileNode{}
+
+// Reconcile reconciles a node and ensures that it has a drain-detection deployment
+// attached to it.
+// The Controller will requeue the Request to be processed again if an error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileNode) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+
+	log.Debugf("reconciling node: %v", request.Name)
+
+	node := &corev1.Node{ObjectMeta: metav1.ObjectMeta{Name: request.Name}}
+
+	err := r.client.Get(context.TODO(), request.NamespacedName, node)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("Could not get node %v", request.NamespacedName)
+	}
+
+	nodeHostnameLabel, ok := node.ObjectMeta.Labels[nodeHostNameKey]
+	if !ok {
+		return reconcile.Result{}, fmt.Errorf("Label key %v does not exist on node %v", nodeHostNameKey, request.NamespacedName)
+	}
+
+	// Create or Update the deployment default/foo
+	deploy := &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("rook-ceph-canary-%v", request.Name),
+			Namespace: r.options.OperatorNamespace,
+		},
+	}
+
+	// CreateOrUpdate the deployment
+	mutateFunc := func() error {
+
+		// lablels for the pod, the deployment, and the deploymentSelector
+		deploymentLabels := map[string]string{
+			nodeHostNameKey: nodeHostnameLabel,
+			"app":           drainDetectionLabel,
+		}
+
+		nodeSelector := map[string]string{nodeHostNameKey: nodeHostnameLabel}
+
+		// Deployment selector is immutable so we set this value only if
+		// a new object is going to be created
+		if deploy.ObjectMeta.CreationTimestamp.IsZero() {
+			deploy.Spec.Selector = &metav1.LabelSelector{
+				MatchLabels: deploymentLabels,
+			}
+		}
+
+		// update
+		deploy.ObjectMeta.Labels = deploymentLabels
+
+		// update the Deployment pod template
+		deploy.Spec.Template = corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{Labels: deploymentLabels},
+			Spec: corev1.PodSpec{
+				NodeSelector: nodeSelector,
+				Containers:   newDoNothingContainers(),
+			},
+		}
+		controllerutil.SetControllerReference(node, deploy, r.scheme)
+
+		return nil
+	}
+	op, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, deploy, mutateFunc)
+
+	if err != nil {
+		log.Error(err, "Node reconcile failed")
+	} else {
+		log.Debug("Deployment successfully reconciled. operation: ", op)
+	}
+	return reconcile.Result{}, nil
+}
+
+// returns a container that does nothing
+func newDoNothingContainers() []corev1.Container {
+	return []corev1.Container{{
+		Image:   "busybox",
+		Name:    "busybox",
+		Command: []string{"bin/sh"},
+		Args:    []string{"-c", "while true; do sleep 3600; done"},
```

Comment:
You don't need the loop, just run `sleep infinity`.

Reply:
thanks!